clientloop.c

```diff
@@ -1599,6 +1599,11 @@ client_request_agent(struct ssh *ssh, const char *request_type, int rchan)
 			debug_fr(r, "ssh_get_authentication_socket");
 		return NULL;
 	}
+	if (ssh->kex->server_keyhashsig != NULL) {
+		if ((r = ssh_agent_bind_hostkey(sock,
```

Comment:
we could just pass 'ssh' to ssh_agent_bind_hostkey(sock, ssh) and have it check ssh->kex itself,
so we only check server_keyhashsig at a single place

Reply:
I don't think we can - authfd.c is linked with ssh-add and ssh-keygen and neither has struct ssh *

Reply:
ok